Makefile

```diff
@@ -8,6 +8,14 @@ all: build
 mocks:
 	$(GO) generate ./...
 
+checkout-enterprise:
+	@if [ ! -d "./enterprise" ]; then \
+	  git clone git@github.com:rudderlabs/rudder-server-enterprise.git enterprise; \
+	fi
+	@if [ -f ".enterprise-commit" ]; then \
+	  cd ./enterprise; git checkout `cat ../.enterprise-commit`; \
+	fi
+
```

Comment:
Just noticed your instructions regarding this usecase in the doc. You can ignore this comment. Also, it's not a bad idea either to ask people to explicitly use this command to update .enterprise-commit file. Just a thought.

Reply:
It doesn't hurt to provide the option to manually update the .enterprise-commit file. There might be scenarios where it could be useful.

Perhaps it best to make the user explicitly update the .enterprise-commit, and have the pre-commit hook issue a warning. My intention was to make the experience as similar as in the submodule scenario, where the hash reference is automatically updated by git, although it is not automatically staged.